src/components/portrait/portrait.test.js

```diff
@@ -0,0 +1,281 @@
+
+      "https://www.gravatar.com/avatar/05c1c705ee45d7ae88b80b3a8866ddaa?s=24&d=404",
+    ])("should check custom image url for Portrait component ", (srcImage) => {
+      CypressMountWithProviders(<PortraitComponent src={srcImage} />);
+      portraitPreview()
+        .should("have.css", "height", "40px")
+        .and("have.css", "width", "40px");
+      portraitPreview().children().should("have.attr", "src", srcImage);
+    });
+
+    it.each([
+      ["XS", 24],
+      ["S", 32],
+      ["M", 40],
+      ["ML", 56],
+      ["L", 72],
+      ["XL", 104],
+      ["XXL", 128],
+    ])(
+      "should check %s size for Portrait component ",
+      (size, heightAndWidth) => {
+        CypressMountWithProviders(<PortraitComponent size={size} />);
+        portraitPreview()
+          .should("have.css", "height", `${heightAndWidth}px`)
+          .and("have.css", "width", `${heightAndWidth}px`);
+      }
+    );
+
+    it.each(testImage)(
+      "should check alt text for %s in Portrait component ",
+      (src) => {
+        CypressMountWithProviders(
+          <PortraitComponent alt="cypress-test" src={src} />
+        );
+        portraitAlt().should("have.attr", "alt", "cypress-test");
+      }
+    );
+
+    it.each([
+      ["square", "0px"],
+      ["circle", "50%"],
+    ])("should check %s shape for Portrait component ", (shape, radius) => {
+      CypressMountWithProviders(<PortraitComponent shape={shape} />);
+      portraitPreview().children().should("have.css", "border-radius", radius);
+    });
+
+    it.each([
+      "filter",
+      "error",
+      "warning",
+      "info",
+      "key",
+      "add",
+      "admin",
+      "alert",
+    ])("should check %s shape for Portrait component ", (iconType) => {
+      CypressMountWithProviders(<PortraitComponent iconType={iconType} />);
+      icon().should("have.attr", "data-element", iconType).and("be.visible");
+    });
+
+    it.each([
+      ["Kate Wool", "KW"],
+      ["Jane Col", "JC"],
+    ])(
+      "should check initials for %s in Portrait component ",
+      (name, passInitials) => {
+        CypressMountWithProviders(
+          <PortraitComponent initials={passInitials} />
+        );
+        portraitInitials()
+          .should("have.css", "height", "38px")
+          .and("have.css", "width", "38px");
+      }
+    );
+
+    it.each([
+      [false, "rgb(242, 245, 246)"],
+      [true, "rgb(153, 173, 183)"],
+    ])(
+      "should check initials for %s in Portrait component ",
+      (boolValue, color) => {
+        CypressMountWithProviders(
+          <PortraitComponent darkBackground={boolValue} />
+        );
+        portraitPreview()
+          .children()
+          .should("have.css", "background-color", color);
+      }
+    );
+
+    describe("should render Portrait component for event tests", () => {
```

Comment:
The event tests should be after the prop tests and other tests. I think this should be placed after the following tooltip tests.